osi_hostvehicledata.proto

```diff
@@ -2,31 +2,281 @@ syntax = "proto2";
+    {
+        // The positions of the pedals.
+        //
+        optional Pedalry pedalry = 1;
+
+        // The actual gear of the transmission.
+        // E.g. a gear lever can be in "D" and transmission in "4", but not the
+        // other way around.
+        //
+        // The sign of this field is linked to the gear's mode as following:
+        // - zero: neutral position
+        // - positive: driving forward mode
+        // - negative: reverse mode (generally -1, but few vehicles have several
+        //   reverse mode gears)
+        //
+        optional int32 gear_transmission = 2;
+
+        // Information about the motor(s).
+        //
+        repeated Motor motor = 3;
+
+        //
+        // \brief A description of the motor states.
+        //
+        message Motor
+        {
+            // The type of the motor.
+            //
+            optional Type type = 1;
+
+            // Rounds per minute of the motor.
+            //
+            // Unit: 1/min
+            //
+            optional double rpm = 2;
+
+            // Torque from the motor.
+            //
+            // Unit: N*m
+            //
+            optional double torque = 3;
+
+            // Definition which type of motor is used.
+            //
+            enum Type
+            {
+                // The motor type is unknown.
+                //
+                TYPE_UNKNOWN = 0;
+
+                // It is another motor type.
+                //
+                TYPE_OTHER = 1;
+
+                // A motor working after the principle of Nicolaus Otto.
+                //
+                TYPE_OTTO = 2;
+
+                // A motor working after the principle of Rudolf Diesel.
+                //
+                TYPE_DIESEL = 3;
+
+                // A motor working electric.
+                //
+                TYPE_ELECTRIC = 4;
+            }
+        }
+    }
+
+    //
+    // \brief The focus here is on the description of the wheels.
+    //
+    message VehicleWheels
+    {
+        // Description of each wheel.
+        //
+        // \note OSI uses singular instead of plural for repeated field names.
+        //
+        repeated WheelData wheel_data = 1;
+
+        //
+        // \brief The focus here is on the description of a wheel.
+        //
+        message WheelData
+        {
+            // The axle which contains this wheel. A value of 0 represents the
+            // front-most axle of the vehicle with higher numbers incrementing
+            // towards the rear-most axle.
+            //
+            optional uint32 axle = 1;
+
+            // The index of the wheel on the axle, counting in the direction
+            // of positive-y, that is, right-to-left.
+            //
+            // For example, on a standard 2-axle, 4-wheel car, the rear-right
+            // wheel would be (axle=1, index=0).
+            // This concept works also for twin tires.
+            //
+            optional uint32 index = 2;
```

Comment:
Maybe add if the respective wheel is driven by a motor

Reply:
What do you exactly mean? Sry, I don't understand the point^^